src/app/app_wrangling.R

```diff
@@ -0,0 +1,204 @@
+        replace_na(list(category = repl_val,
+                        mechanic = repl_val,
+                        publisher = repl_val)) %>%
+        mutate(category = strsplit(category, reg_use),
+               mechanic = strsplit(mechanic, reg_use),
+               publisher = strsplit(publisher, reg_use)) %>%
+        select(-X1)
+
+    return(boardgame_data)
+}
+
+# Filters the dataframe and returns the top n
+call_boardgame_filter <- function(data, cat, mech, pub, n = NULL) {
+    func_df_out <- data
+
+    # check each category
+    cat_check <- call_bool_series_and(data, "category", cat)
+    mech_check <- call_bool_series_and(data, "mechanic", mech)
+    publ_check <- call_bool_series_and(data, "publisher", pub)
+
+    # remove rows from df that aren't matched
+    func_df_out <- func_df_out %>%
+        mutate(cat_check, mech_check, publ_check) %>%
+        filter(cat_check, mech_check, publ_check) %>%
+        select(-cat_check, -mech_check, -publ_check) %>%
+        # sort by average rating
+        arrange(desc(average_rating))
+
+    # return number of entries is specified by user
+    if (!is.null(n)) {
+        func_df_out <- slice_head(func_df_out, n = n)
+    }
+
+    return(func_df_out)
+}
+
+
+# helper function to check where user input matches
+# used in the `*and()` and `*or()` functions
+check_list <- function(col_data, list_) {
+    map(col_data, ~ (list_ %in% unlist(.x)))
+}
+
+
+# function which checks if conditions are met for all requirements
+call_bool_series_and <- function(data, col_, list_) {
+
+    # check if all of the values contain the user input list
+    check <- check_list(data[[col_]], list_) %>%
+        map(~all(.)) %>%
+        unlist()
+
+    # if there are no TRUE values in the entire list, switch all values to TRUE
+    if (sum(check) == 0) check <- !check
+    return(check)
+}
+
+
+# function which checks if one condition is met from requirements
+call_bool_series_or <- function(data, col_, list_) {
+
+    # check if one of the values matches the user input list
+    check <- check_list(data[[col_]], list_) %>%
+        map(~any(.)) %>%
+        unlist()
+
+    return(check)
+}
+
+
+# call from tab 1 of graph to filter data based on user selection
+call_boardgame_radio <- function(data, col_, list_) {
+    func_df_out <- data
+    # subset based on user selection
+    func_df_out <- func_df_out[call_bool_series_or(data, col_, list_), ]
+    # call form group to add group column
+    func_df_out <- form_group(func_df_out, col_, list_)
+    # remove all entries that aren't part of a group
+    fucn_df_out <- filter(func_df_out, !is.na(group))
+    return(fucn_df_out)
+}
+
+
+# used in form group map call to check what should be assigned
+form_group_helper <- function(data, list_) {
+    if (length(data) == 0) {
+        return(NA)
+    }
+    else if (length(list_) > 1 & all(data)) {
+        return("All Selected")
+    }
+    else {
+        return(list_[data])
+    }
```

Comment:
Not exactly sure if I follow this.  I'll just have to check it out when it's merged in